### ISSUE TITLE
Fix "leaking" parallel tests

### ISSUE THREAD

**Issue by devinrsmith** [deephaven/deephaven-core#643]
Nightly CI has been consistently failing due to workflow timeouts (being addressed in #605).

Part of that investigation noticed that "parallel" tests are consistently the 6 hour+ culprit; but by making the "parallel" tests each start their own VM (like the "serial" tests), the tests complete consistently in ~4 hours. This likely means that the "parallel" tests have some sort of leak, causing test runs in the same VM to take a lot more time. It may be a memory leak, or other logical test-structural issues (deadlock, slowlock, etc).

To get CI working, the "parallel" tests were switched to be functionally the same as the serial tests (in logical setup, but still separate tasks).

Note: "parallel" tests in the context of our current GH Actions setup is a bit misleading, as we don't actually enable parallelism.

**Comment by rcaudy** [comment]
This may simply be heap fragmentation, which can be addressed with `System.gc()`. That said, if we can make it so the runner will clear the query scope, query library, and compiler tools context between runs it may help.

**Comment by devinrsmith** [comment]
It seems like the leak / timeouts perpetually seen in testParallel was solved by #644. (That said, nightly ci job has also been re-structured to execute different test tasks in separate jobs, to speed up the process.)

**Comment by jcferretti** [comment]
I did 74 consecutive runs of the short version of testParallel without failures.  So it would seem like the leaks either were resolved, or are now low probability.

Gradle options like so:

```
#!/bin/bash

set -ex

if [ $# -gt 1 ]; then
  echo "Usage: $0 [iterations]" 1>&2
  exit 1
fi

if [ $# -eq 1 ]; then
  max=$1
else
  max=10000
fi

echo $max
exit 0

j=0

while [ $j -lt $max ]; do
    echo "== Start run $j `date` =="
    ./gradlew cleanTestParallel 
    ./gradlew \
        --console=plain \
        --no-build-cache \
        :engine-table:testParallel \
        -PallowParallel=true \
        -PmaxParallelForks=80 \
        -PforkEvery=10 \
        -PmaxHeapSize=30g \
        -PshortTests=true \
        --parallel \
        --max-workers=80
    if [ $? -ne 0 ]; then
       exit 1
    fi
    echo "== End run $j `date` =="
    echo
    j=$((j + 1))
done
```
